apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/loader/AgentClassLoader.java

```diff
@@ -169,31 +135,15 @@ private Class<?> processLoadedClass(Class<?> loadedClass) {
         return loadedClass;
     }
 
-    private List<Jar> getAllJars() {
-        if (allJars == null) {
-            jarScanLock.lock();
-            try {
-                if (allJars == null) {
-                    allJars = doGetJars();
-                }
-            } finally {
-                jarScanLock.unlock();
-            }
-        }
-
-        return allJars;
-    }
-
-    private LinkedList<Jar> doGetJars() {
-        LinkedList<Jar> jars = new LinkedList<>();
+    private static LinkedList<URL> doGetJars(List<File> classpath) {
```

Comment:
Why is this static?

Reply:
Because it needs to be called in a static method, and only need to initialize the plugin list once